gwpy/timeseries/timeseries.py

```diff
@@ -1463,18 +1474,26 @@ def taper(self, side='leftright'):
         if side not in ('left', 'right', 'leftright'):
             raise ValueError("side must be one of 'left', 'right', "
                              "or 'leftright'")
+        if duration and nsamples:
+            raise ValueError("only one of 'duration' or 'nsamples' may be "
+                             "provided as arguments")
+        elif duration:
```

Comment:
@tjma12, instead of this `elif` block, it might appease Codeclimate if you use two pythonic shorthands:

```python
    duration = duration or 0
    nsamples = nsamples or int(duration * self.sample_rate.to("Hz").value)
```

The first line sets `duration` to 0 if it isn't passed, the second does the same for `nsamples`, and this plays nicely with the `if not nsamples` block on line 1487.

Reply:
I actually think the code as it exists now is easier to read, regardless of what codeclimate will say. I am also not convinced that this _will_ appease codeclimate, since there are still conditionals.

Reply:
I left the check on `duration` in place and noted in the docstring that it will override `nsamples` if both are provided